ui/apps/platform/src/hooks/useFetchNamespaceDeployments.ts

```diff
@@ -1,68 +1,87 @@
-import { useState, useEffect } from 'react';
-import { gql, useQuery } from '@apollo/client';
+import { useState } from 'react';
+import useDeepCompareEffect from 'use-deep-compare-effect';
 
-import queryService from 'utils/queryService';
+import { listDeployments } from 'services/DeploymentsService';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
 
 type Deployment = {
     id: string;
     name: string;
 };
 export type NamespaceWithDeployments = {
     metadata: {
-        id: string;
         name: string;
     };
     deployments: Deployment[];
 };
-type DeploymentResponse = {
-    results: NamespaceWithDeployments[];
-};
 
-const DEPLOYMENTS_FOR_NAMESPACE_QUERY = gql`
-    query getNamespaceDeployments($query: String!) {
-        results: namespaces(query: $query) {
-            metadata {
-                name
-                id
-            }
-            deployments {
-                name
-                id
-            }
-        }
-    }
-`;
+type ListDeploymentResponse = {
+    loading: boolean;
+    error: string;
+    deploymentsByNamespace: NamespaceWithDeployments[];
+};
 
 function useFetchNamespaceDeployments(selectedNamespaceIds: string[]) {
-    const [availableDeployments, setAvailableNamespaces] = useState<NamespaceWithDeployments[]>([]);
-
-    const searchClause = { 'Namespace ID': selectedNamespaceIds };
-    // If the selectedNamespaceId has not been set yet, do not run the gql query
-    const queryOptions =
-        selectedNamespaceIds.length > 0
-            ? { variables: { query: queryService.objectToWhereClause(searchClause) } }
-            : { skip: true };
+    const [deploymentResponse, setDeploymentResponse] = useState<ListDeploymentResponse>({
+        loading: false,
+        error: '',
+        deploymentsByNamespace: [],
+    });
+    useDeepCompareEffect(() => {
+        if (selectedNamespaceIds.length > 0) {
```

Comment:
1. Although I understand why not to make a request if the id array is empty, it is not clear to me whether or not whether there needs to be a change to value of `deploymentsByNamespace` array. My intuitive impression is that it should become `[]` like the initial value, but I do not see that in the GraphQL nor REST version.
2. If that observation is correct, it suggests that the calling code is written not to depends on the hook clearing the deployments if the user clears the namespace selections.
3. If that hypothesis is correct, then here are 2 questions:
    * Thinking of the hook as reusable for other situations, what should its behavioral contract be for a change from some namespaces selected to no namespaces selected?
    * Would a change in behavior break the existing use of the hook?